src/Operations.jl

```diff
@@ -597,22 +598,36 @@ end
 function download_artifacts(pkg_roots::Vector{String}; platform::AbstractPlatform=HostPlatform(),
                             verbose::Bool=false)
     # List of Artifacts.toml files that we're going to download from
-    artifacts_tomls = String[]
+    artifacts_tomls = Tuple{String,Dict}[]
 
     for path in pkg_roots
         # Check to see if this package has an (Julia)Artifacts.toml
         for f in artifact_names
             artifacts_toml = joinpath(path, f)
             if isfile(artifacts_toml)
-                push!(artifacts_tomls, artifacts_toml)
+                hook_path = joinpath(path, ".pkg", "select_artifacts_hook.jl")
+
+                if isfile(hook_path)
+                    mktempdir() do project
+                        meta_toml = String(read(`$(Base.julia_cmd()) --project=$(project) $(hook_path)`))
+                        push!(artifacts_tomls, (artifacts_toml, TOML.parse(meta_toml)))
```

Comment:
In the future, it would be good if we had a typed `struct` that we parse the Artifact.toml into, instead of passing around the raw dictionary from `TOML.parse`.